src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,482 @@
+    }
+
+    /**
+     * Gets whether or not this component is in italics.
+     *
+     * @return True iff it is.
+     */
+    public boolean isItalic() {
+        return italic != null && italic.get();
+    }
+
+    /**
+     * Sets this component's italic.
+     *
+     * @param italic The italic.
+     *
+     * @return This component.
+     */
+    public ChatComponent setItalic(boolean italic) {
+        $lock.writeLock().lock();
+        if (this.italic == null)
+            this.italic = new AtomicBoolean(italic);
+        else
+            this.italic.set(italic);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is underlined.
+     *
+     * @return True iff it is.
+     */
+    public boolean isUnderlined() {
+        return underlined != null && underlined.get();
+    }
+
+    /**
+     * Sets this component's underlined.
+     *
+     * @param underlined The underline.
+     *
+     * @return This component.
+     */
+    public ChatComponent setUnderlined(boolean underlined) {
+        $lock.writeLock().lock();
+        if (this.underlined == null)
+            this.underlined = new AtomicBoolean(underlined);
+        else
+            this.underlined.set(underlined);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is striked through.
+     *
+     * @return True iff it is.
+     */
+    public boolean isStrikethrough() {
+        return strikethrough != null && strikethrough.get();
+    }
+
+    /**
+     * Sets this component's strikethrough.
+     *
+     * @param strikethrough The strikethrough.
+     *
+     * @return This component.
+     */
+    public ChatComponent setStrikethrough(boolean strikethrough) {
+        $lock.writeLock().lock();
+        if (this.strikethrough == null)
+            this.strikethrough = new AtomicBoolean(strikethrough);
+        else
+            this.strikethrough.set(strikethrough);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is in obfuscation.
+     *
+     * @return True iff it is.
+     */
+    public boolean isObfuscated() {
+        return obfuscated != null && obfuscated.get();
+    }
+
+    /**
+     * Sets this component's obfuscation.
+     *
+     * @param obfuscated The new obfuscated state.
+     *
+     * @return This component.
+     */
+    public ChatComponent setObfuscated(boolean obfuscated) {
+        $lock.writeLock().lock();
+        if (this.obfuscated == null)
+            this.obfuscated = new AtomicBoolean(obfuscated);
+        else
+            this.obfuscated.set(obfuscated);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets this component as a JSON object, ready to be sent to a client.
+     *
+     * @return The JSON object.
+     */
+    public JsonObject asJson() {
+        $lock.readLock().lock();
```

Comment:
THREAD-SAFETY
As long as the rest of the class is thread safe, aggregating once modification has finished should not need to lock as the fields are correctly updated and follows happens before order.